src/tests/unit/connection.test.js

```diff
@@ -0,0 +1,79 @@
+import chai from 'chai';
+import chaiExclude from 'chai-exclude';
+import chaiAsPromised from 'chai-as-promised';
+import config from 'config';
+import proxyquireModule from 'proxyquire';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+// Prevent call thru to original dependencies
+const proxyquire = proxyquireModule.noCallThru();
+
+chai.should();
+chai.use(chaiExclude);
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+
+afterEach(() => { sinon.restore(); });
+
+describe('Test oracledb connection module', () => {
+  let configGetStub;
+  let connection;
+
+  beforeEach(() => {
+    configGetStub = sinon.stub(config, 'get')
+      .withArgs('dataSources.oracledb')
+      .returns({});
+  });
+
+  const createOracleDbStub = (stubs) => {
+    connection = proxyquire('api/v1/db/oracledb/connection', {
+      config: { get: configGetStub },
+      oracledb: stubs,
+      // suppress logger output for testing
+      '../../../../utils/logger': { logger: { error: () => {} } },
+    });
+  };
+
+  describe('getConnection', () => {
+    it('Should call createPool if pool is falsy. Should not call createPool additional times', async () => {
+      const createPoolStub = sinon.stub()
+        .resolves({ getConnection: async () => 'test-connection' });
+      createOracleDbStub({ createPool: createPoolStub });
```

Comment:
Looks like every time `createOracleDbStub` is called `createPool` is always the function being stubbed. You should be able to simplify this by just passing in `createPoolStub` and changing `oracledb: stubs,` to `oracledb: { createPool: stubs },`